### PR TITLE
Accept any RMW implementation, not just the default

### PR DESCRIPTION
**Pull request by cottsay**
The `get_default_rmw_implementation` macro contains the same functionality as is removed by this change, but it contains logic to fall back to any available RMW implementation if neither `RMW_IMPLEMENTATION` is specified nor the default is available.

### CODE REVIEW

**Comment by cottsay** [comment]
Here's some pretty broad CI jobs to check for regressions, but it should be noted that the crux of the problem that this change addresses can't be tested on ci.ros2.org because of the way the jobs are structured.
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13539/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8436/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11260/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13607/)

**Comment by clalancette** [comment]
The idea here looks good to me, and solved the problem in my local testing.  So I'm somewhat confident it will fix the underlying problem.  It looks like the `test_rmw_implementation` test is failing for some reason, though.

**Comment by cottsay** [comment]
> It looks like the `test_rmw_implementation` test is failing for some reason, though.

I just suck at package selection arguments. Trying again:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13540/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8437/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11261/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13608/)

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
@cottsay Once you merge this, can you please do a new release of rmw_implementation into Rolling?  That should unblock the build of demo_nodes_cpp_native.  Thanks.

**Pull request merged**
